### ISSUE TITLE
Big companies v. startups

### ISSUE THREAD

**Issue by guevara** [guevara/read-it-later#5017]
Big companies v. startups<br>
<br>
https://ift.tt/2bJXytE<br>
<br>
<br>
<br>
<div>
<strong>Big companies v. startups</strong>
<hr>
<p>There's a meme that's been going around for a while now: you should join a startup because the money is better and the work is more technically interesting. <a href="http://paulgraham.com/wealth.html">Paul Graham says</a> that the best way to make money is to "start or join a startup", which has been "a reliable way to get rich for hundreds of years", and that you can "compress a career's worth of earnings into a few years". Michael Arrington says that you'll <a href="https://www.jwz.org/blog/2011/11/watch-a-vc-use-my-name-to-sell-a-con/">become a part of history</a>. Joel Spolsky says that by joining a big company, you'll end up <a href="http://www.joelonsoftware.com/items/2008/05/01.html">playing foosball and begging people to look at your code</a>. Sam Altman says that if you join Microsoft, <a href="http://blog.samaltman.com/advice-for-ambitious-19-year-olds">you won't build interesting things and may not work with smart people</a>. They all claim that you'll learn more and have better options if you go work at a startup. Some of these links are a decade old now, but the same ideas are still circulating and those specific essays are still cited today.</p>
<p>Let's look at these points one one-by-one.</p>
<ol>
<li>You'll earn much more money at a startup</li>
<li>You won't do interesting work at a big company</li>
<li>You'll learn more at a startup and have better options afterwards</li>
</ol>
<h3>1. Earnings</h3>
<p>The numbers will vary depending on circumstances, but we can do a back of the envelope calculation and adjust for circumstances afterwards. Median income in the U.S. is <a href="https://en.wikipedia.org/wiki/Personal_income_in_the_United_States">about $30k/yr</a>. The somewhat bogus zeroth order lifetime earnings approximation I'll use is $30k * 40 = $1.2M. A new grad at Google/FB/Amazon with a lowball offer will have a total comp (salary + bonus + equity) of $130k/yr. <a href="http://www.glassdoor.com/Salary/Google-Senior-Software-Engineer-Salaries-E9079_D_KO7,31.htm">According to glassdoor's current numbers</a>, someone who makes it to T5/senior at Google should have a total comp of around $250k/yr. These are fairly conservative numbers.</p>
<p>Someone who's not particularly successful, but not particularly unsuccessful will probably make senior in five years. For our conservative baseline, let's assume that we'll never make it past senior, into the pay grades where compensation really skyrockets. We'd expect earnings (total comp including stock, but not benefits) to looks something like:</p>
<table>
<thead>
<tr>
<th>Year</th>
<th>Total Comp</th>
<th>Cumulative</th>
</tr>
</thead>
<tbody>
<tr>
<td>0</td>
<td>130k</td>
<td>130k</td>
</tr>
<tr>
<td>1</td>
<td>160k</td>
<td>290k</td>
</tr>
<tr>
<td>2</td>
<td>190k</td>
<td>480k</td>
</tr>
<tr>
<td>3</td>
<td>220k</td>
<td>700k</td>
</tr>
<tr>
<td>4</td>
<td>250k</td>
<td>950k</td>
</tr>
<tr>
<td>5</td>
<td>250k</td>
<td>1.2M</td>
</tr>
<tr>
<td>&hellip;</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<td>9</td>
<td>250k</td>
<td>2.2M</td>
</tr>
<tr>
<td>&hellip;</td>
<td>&hellip;</td>
<td>&hellip;</td>
</tr>
<tr>
<td>39</td>
<td>250k</td>
<td>9.7M</td>
</tr>
</tbody>
</table>
<p>Looks like it takes six years to gross a U.S. career's worth of income. If you want to adjust for the increased tax burden from earning a lot in a few years, add an extra year. Maybe add one to two more years if you decide to live in the bay or in NYC. If you decide not to retire, lifetime earnings for a 40 year career comes in at almost $10M.</p>
<p>One common, but false, objection to this is that your earnings will get eaten up by the cost of living in the bay area. Not only is this wrong, it's actually the opposite of correct. You can work at these companies from outside the bay area; most of these companies will pay you maybe 10% less if you work in a location where cost of living is around the U.S. median by working in a satellite office of a trendy company headquartered in SV or Seattle (at least if you work in the US -- pay outside of the US is often much lower for reasons that don't really make sense to me). Market rate at smaller companies in these areas tends to be very low. When I interviewed in places like Portland and Madison, there was a 3x-5x difference between what most small companies were offering and what I could get at a big company in the same city. In places like Austin, where the market is a bit thicker, it was a 2x-3x difference. The difference in pay at 90%-ile companies is greater, not smaller, outside of the SF bay area.</p>
<p>Another objection is that <a href="http://www.bls.gov/ooh/computer-and-information-technology/computer-programmers.htm">most programmers at most companies don't make this kind of money</a>. If, three or four years ago, you'd told me that there's a career track where it's totally normal to make $250k/yr after a few years, doing work that was fundamentally pretty similar to the work I was doing then, I'm not sure I would have believed it. No one I knew made that kind of money, except maybe the CEO of the company I was working at. Well him, and folks who went into medicine or finance.</p>
<p>The only difference between then and now is that I took a job at a big company. When I took that job, the common story I heard at orientation was basically &ldquo;I never thought I'd be able to get a job at Google, but a recruiter emailed me and I figured I might as well respond&rdquo;. For some reason, women were especially likely to have that belief. Anyway, I've told that anecdote to multiple people who didn't think they could get a job at some trendy large company, who then ended up applying and getting in. And what you'll realize if you end up at a place like Google is that most of them are just normal programmers like you and me. If anything, I'd say that Google is, on average, less selective than the startup I worked at. When you only have to hire 100 people total, and half of them are folks you worked with as a technical fellow at one big company and then as an SVP at another one, you can afford to hire very slowly and being extremely selective. Big companies will hire more than 100 people per week, which means they can only be so selective.</p>
<p>Despite the hype about how hard it is to get a job at Google/FB/wherever, your odds aren't that bad, and <a href="http://www.bloomberg.com/news/articles/2015-12-17/big-ipo-tiny-payout-for-many-startup-workers?trk=pulse-det-art_view_ext">they're certainly better than your odds striking it rich at a startup</a>, for which <a href="http://www.kalzumeus.com/2011/10/28/dont-call-yourself-a-programmer/">Patrick McKenzie has a handy cheatsheet</a>:</p>
<blockquote>
<p>Roll d100. (Not the right kind of geek? Sorry. rand(100) then.)<br>
<strong>0~70</strong>: Your equity grant is worth nothing.<br>
<strong>71~94</strong>: Your equity grant is worth a lump sum of money which makes you about as much money as you gave up working for the startup, instead of working for a megacorp at a higher salary with better benefits.<br>
<strong>95~99</strong>: Your equity grant is a life changing amount of money. You won't feel rich &mdash; you're not the richest person you know, because many of the people you spent the last several years with are now richer than you by definition &mdash; but your family will never again give you grief for not having gone into $FAVORED_FIELD like a proper $YOUR_INGROUP.<br>
<strong>100</strong>: You worked at the next Google, and are rich beyond the dreams of avarice. Congratulations.<br>
Perceptive readers will note that 100 does not actually show up on a d100 or rand(100).</p>
</blockquote>
<p>For a more serious take that gives approximately the same results, <a href="https://80000hours.org/2014/05/how-much-do-y-combinator-founders-earn/">80000 hours finds that the average value of a YC founder after 5-9 years is $18M</a>. That sounds great! But there are a few things to keep in mind here. First, YC companies are unusually successful compared to the average startup. Second, in their analysis, 80000 hours notes that 80% of the money belongs to 0.5% of companies. Another 22% are worth enough that founder equity beats working for a big company, but that leaves 77.5% where that's not true.</p>
<p>If you're an employee and not a founder, the numbers look a lot worse. If you're a very early employee you'd be quite lucky to get 1/10th as much equity as a founder. If we guess that 30% of YC startups fail before hiring their first employee, that puts the mean equity offering at $1.8M / .7 = $2.6M. That's low enough that for 5-9 years of work, you really need to be in the 0.5% for the payoff to be substantially better than working at a big company unless the startup is paying a very generous salary.</p>
<p>There's a sense in which these numbers are too optimistic. Even if the company is successful and has a solid exit, there are plenty of things that can make your equity grant worthless. It's hard to get statistics on this, but anecdotally, this seems to be the common case in acquisitions.</p>
<p>Moreover, the pitch that you'll only need to work for four years is usually untrue. To keep your <a href="https://danluu.com/startup-options/">lottery ticket</a> until it pays out (or fizzles out), you'll probably have to stay longer. The most common form of equity at early stage startups are <a href="https://en.wikipedia.org/wiki/Incentive_stock_option">ISOs</a> that, by definition, expire 90 at most days after you leave. If you get in early, and leave after four years, you'll have to exercise your options if you want a chance at the lottery ticket paying off. If the company hasn't yet landed a large valuation, you might be able to get away with paying O(median US annual income) to exercise your options. If the company looks like a rocket ship and VCs are piling in, you'll have a massive tax bill, too, all for a lottery ticket.</p>
<p>For example, say you joined company X early on and got options for 1% of the company when it was valued at $1M, so the cost exercising all of your options is only $10k. Maybe you got lucky and four years later, the company is valued at $1B and your options have only been diluted to .5%. Great! For only $10k you can exercise your options and then sell the equity you get for $5M. Except that the company hasn't IPO'd yet, so if you exercise your options, you're stuck with a tax bill from making $5M, and by the time the company actually has an IPO, your stock could be worthy anywhere from $0 to $LOTS. In some cases, you can sell your non-liquid equity for some fraction of its &ldquo;value&rdquo;, but my understanding is that it's getting more common for companies to add clauses that limit your ability to sell your equity before the company has an IPO. And even when your contract doesn't have a clause that prohibits you from selling your options on a secondary market, <a href="https://news.ycombinator.com/item?id=10705646">companies sometimes use backchannel communications to keep you from being able to sell your options</a>.</p>
<p>Of course not every company is like this -- I hear that Dropbox has generously offered to buy out people's options at their current valuation for multiple years running and they now hand out RSUs instead of options, and Pinterest now gives people seven years to exercise their options after they leave -- but stories like that are uncommon enough that they're notable. The result is that people are incentivized to stay at most startups, even if they don't like the work anymore. From chatting with my friends at well regarded highly-valued startups, it sounds like many of them have a substantial fraction of zombie employees who are just mailing it in and waiting for a liquidity event. A common criticism of large companies is that they've got a lot of lifers who are mailing it in, but most large companies will let you leave any time after the first year and walk away with a pro-rated fraction of your equity package. It's startups where people are incentivized to stick around even if they don't care about the job.</p>
<p>At a big company, we have a career's worth of income in six years with high probability once you get your foot in the door. This isn't quite as good as the claim that you'll be able to do that in three or four years at a startup, but the risk at a big company is very low once you land the job. In startup land, we have a lottery ticket that appears to have something like a 0.5% chance of paying off for very early employees. Startups might have had a substantially better expected value when Paul wrote about this in 2004, but big company compensation has increased much faster than compensation at the median startup. We're currently in the best job market the world has ever seen for programmers. That's likely to change at some point. The relative returns on going the startup route will probably look a lot better once things change, but for now, saving up some cash while big companies hand it out like candy doesn't seem like a bad idea.</p>
<h3>2. Interesting work</h3>
<p>We've established that big companies will pay you decently. But there's more to life than making money. After all, you spend 40+ hours a week working. How interesting is the work at big companies? Joel claimed that large companies don't solve interesting problems and that Google is <a href="http://www.joelonsoftware.com/items/2008/05/01.html">paying untenable salaries to kids with more ultimate frisbee experience than Python, whose main job will be to play foosball in the googleplex</a>, Sam Altman said something similar (but much more measured) about Microsoft, every third Michael O. Church comment is about how Google tricks a huge number of overqualified programmers into taking jobs that no one wants, and basically every advice thread on HN or reddit aimed at new grads will have multiple people chime in on how the experience you get at startups is better than the experience you'll get slaving away at a big company.</p>
<p>The claim that big companies have boring work is too broad and absolute to even possibly be true. It depends on what kind of work you want to do. When I look at conferences where I find a high percentage of the papers compelling, the stuff I find to be the most interesting is pretty evenly split between big companies and academia, with the (very) occasional paper by a startup. For example, looking at ISCA this year, there's a 2:1 ratio of papers from academia to industry (and all of the industry papers are from big companies). But looking at the actual papers, a significant fraction of the academic papers are reproducing unpublished work that was done at big companies but not published, sometimes multiple years ago. If I only look at the new work that I'm personally interested in, it's about a 1:1 ratio. There are some cases where a startup is working in the same area and not publishing, but that's quite rare and large companies do much more research that they don't publish. I'm just using papers as a proxy for having the kind of work I like. There are also plenty of areas where publishing isn't the norm, but large companies do the bulk of the cutting edge work.</p>
<p>Of course YMMV here depending on what you want to do. I'm not really familiar with the landscape of front-end work, but it seems to me that big companies don't do the vast majority of the cutting edge non-academic work, the way they do with large scale systems. IIRC, there's an HN comment where Jonathan Tang describes how he created his own front-end work: he had the idea, told his manager about it, and got approval to make it happen. It's possible to do that kind of thing at a large company, but people often seem to have an easier time pursuing that kind of idea at a small company. And if your interest is in product, small companies seem like the better bet (though, once again, I'm pretty far removed from that area, so my knowledge is secondhand).</p>
<p>But if you're interested in large systems, at both of my last two jobs, I've seen speculative research projects with 9 figure pilot budgets approved. In a pitch for one of the products, the pitch wasn't even that the project would make the company money. It was that a specific research area was important to the company, and that this infrastructure project would enable the company to move faster in that research area. Since the company is a $X billion dollar a year company, the project only needed to move the needle by a small percentage to be worth it. And so a research project whose goal was to speed up the progress of another research project was approved. Startups simply don't have the resources to throw that much money at research problems that aren't core to their business. And many problems that would be hard problems at startups are curiosities at large companies. Work at Google and have a question that requires running a query that takes 10k machines? No problem! But that's basically impossible to do at a startup, not even considering the fact that you can run the query across data startups can't possibly get.</p>
<p>The flip side of this is that there are experiments that startups have a very easy time doing that established companies can't do. When I was at <a href="http://www.sigecom.org/">EC</a> a number of years ago, back when Facebook was still relatively young, the Google ad auction folks remarked to the FB folks that FB was doing the sort of experiments they'd do if they were small enough to do them, but they couldn't just change the structure of their ad auctions now that there was so much money flowing through their auctions. As with everything else we're discussing, there's a trade-off here and the real question is how to weight the various parts of the trade-off, not which side is better in all ways.</p>
<p>The Michael O. Church claim is somewhat weaker: big companies have cool stuff to work on, but you won't be allowed to work on them until you've paid your dues working on boring problems. A milder phrasing of this is that getting to do interesting work is a matter of getting lucky and landing on an initial project you're interested in, but the key thing here is that most companies can give you a pretty good estimate about how lucky you're going to be. Google is notorious for its blind allocation process, and I know multiple people who ended up at MS because they had the choice between a great project at MS and blind allocation at Google, but even Google has changed this to some extent and it's not uncommon to be given multiple team options with an offer. In that sense, big companies aren't much different from startups. It's true that there are some startups that will basically only have jobs that are interesting to you (e.g., <a href="https://faunadb.com/">FaunaDB</a> if you're interested in building a distributed database). But at any startup that's bigger and less specialized, there's going to be work you're interested in and work you're not interested in, and it's going to be up to you to figure out if your offer lets you work on stuff you're interested in.</p>
<p>Something to note is that if, per (1), you have the leverage to negotiate a good compensation package, you also have the leverage to negotiate for work that you want to do. We're in what is probably the best job market for programmers ever. That might change tomorrow, but until it changes, you have a lot of power to get work that you want.</p>
<p>If this sounds completely foreign to you and you don't have that kind of leverage, I understand. That was me a few years ago. Taking a job at a trendy big company is one way to get that leverage. Companies really want you to make it to &ldquo;senior engineer&rdquo; (where total comp starts at $250k to $350k, depending on the company); hiring is very expensive for them and they're heavily incentivized to mentor the people they hire until they're valuable and productive. Some companies are better at this than others, but the average big company that people want to work for has a lot more resources devoted to helping people learn than almost any startup. The goal at most big companies to get everyone to the senior level. Of course they'll keep hiring which means there will always be non-senior people, but the definition of senior engineer is basically someone who can independently find and solve problems and doesn't require any handholding, i.e., someone who's easy to scale horizontally. Google even has a (unevenly enforced) policy that people who don't "eventually" get to senior should be managed out, and you'll notice that they're not known for having a high involuntary attrition rate. They, and most other big companies, take teaching seriously.</p>
<h3>3. Learning / Experience</h3>
<p>What about the claim that experience at startups is more valuable? We don't have the data to do a rigorous quantitative comparison, but qualitatively, everything's on fire at startups, and you get a lot of breadth putting out fires, but you don't have the time to explore problems as deeply.</p>
<p>I spent the first seven years of my career at a startup and I loved it. It was total chaos, which gave me the ability to work on a wide variety of different things and take on more responsibility than I would have gotten at a bigger company. I did everything from add fault tolerance to an in-house distributed system to owning a quarter of a project that added ARM instructions to an x86 chip, creating both the fastest ARM chip at the time, as well as the only chip capable of switching between ARM and x86 on the fly. That was a great learning experience.</p>
<p>But I've had great learning experiences at big companies, too. At Google, my &ldquo;starter&rdquo; project was to join a previously one-person project, read the half finished design doc, provide feedback, and then start implementing. The impetus for the project was that people were worried that a certain class of applications would require Google to double the number of machines it owns if a somewhat unlikely but not impossible scenario happened. That wasn't too much different from my startup experience, except for that bit about actually having a design doc, and that cutting infra costs could save billions a year instead of millions a year.</p>
<p>The next difference was that, at some point, people way above my pay grade made the decision to get serious about the project, and a lot of high-powered people ended up getting brought in to work on the project or at least provide input, folks like <a href="https://scholar.google.com/scholar?hl=en&amp;q=norman+p+jouppi&amp;btnG=&amp;as_sdt=1%2C48&amp;as_sdtp=">Norm Jouppi</a>, <a href="https://scholar.google.com/citations?user=JicYPdAAAAAJ&amp;hl=en&amp;oi=ao">Geoff Hinton</a>, and <a href="https://scholar.google.com/citations?user=NMS69lQAAAAJ&amp;hl=en&amp;oi=ao">Jeff Dean</a>.</p>
<p>Was that project a better or worse learning experience than the equivalent project at a startup? At a startup, the project probably would have continued to be a two-person show, and I would have learned all the things you learn when you bang out a project with not enough time and resources and do half the thing yourself. Instead, I ended up owning a fraction of the project and merely provided feedback on the rest, and it was merely a matter of luck (timing) that I had significant say on fleshing out the architecture. I definitely didn't get the same level of understanding I would have if I implemented half of it myself. On the other hand, the larger team meant that we actually had time to do things like design reviews and code reviews, and I got feedback from people who have way more experience and knowledge than me. My experience at MS is similar -- I only own maybe a quarter of the project I'm working on, and there's an architect above me who's well regarded and will get most of the credit if the project works out. But when I had a question the other day, I emailed a Turing award winner and got a response back within an hour. It's almost impossible to have access to the same breadth and depth of expertise at a startup. As a result, there are things I've learned in an hour long design review that it would have taken me months or years to learn if I was implementing things myself.</p>
<p>If you care about impact, it's also easier to have a large absolute impact at a large company, due to the scale that big companies operate at. If I implemented what I'm doing now for a companies the size of the startup I used to work for, it would have had an impact of maybe $10k/month. That's nothing to sneeze at, but it wouldn't have covered my salary. But the same thing at a big company is worth well over 1000x that. There are simply more opportunities to have high impact at large companies because they operate at a larger scale. The corollary to this is that startups are small enough that it's easier to have an impact on the company itself, even when the impact on the world is smaller in absolute terms. Nothing I do is make or break for a large company, but when I worked at a startup, it felt like what we did could change the odds of the company surviving.</p>
<p>As far as having better options after having worked for a big company or having worked for a startup, if you want to work at startups, you'll probably have better options with experience at startups. If you want to work on the sorts of problems that are dominated by large companies, you're better off with more experience in those areas, at large companies. There's no right answer here.</p>
<h3>Conclusion</h3>
<p>The compensation trade-off has changed a lot over time. When Paul Graham was writing in 2004, he used $80k/yr as a reasonable baseline for what &ldquo;a good hacker&rdquo; might make. Adjusting for inflation, <a href="http://data.bls.gov/cgi-bin/cpicalc.pl?cost1=80000&amp;year1=2004&amp;year2=2015">that's about $100k/yr now</a>. But the total comp for &ldquo;a good hacker&rdquo; is $250k+/yr, not even counting perks like free food and having really solid insurance. The trade-off has heavily tilted in favor of large companies.</p>
<p>The interesting work trade-off has also changed a lot over time, but the change has been&hellip; bimodal. The existence of AWS and Azure means that ideas that would have taken millions of dollars in servers and operational expertise can be done with almost no fixed cost and low marginal costs. The scope of things you can do at an early-stage startup that were previously the domain of well funded companies is large and still growing. But at the same time, if you look at the work Google and MS are publishing at top systems conferences, startups are farther from being able to reproduce the scale-dependent work than ever before (and a lot of the most interesting work doesn't get published). Depending on what sort of work you're interested in, things might look relatively better or relatively worse at big companies.</p>
<p>In any case, the reality is that the difference between types of companies is smaller than the differences between companies of the same type. That's true whether we're talking about startups vs. big companies or mobile gaming vs. biotech. This is recursive. The differences between different managers and teams at a company can easily be larger than the differences between companies. If someone tells you that you should work for a certain type of company, that advice is guaranteed to be wrong much of the time, whether that's a VC advocating that you should work for a startup or a Turing award winner telling you that you should work in a research lab.</p>
<p>As for me, well, I don't know you and it doesn't matter to me whether you end up at a big company, a startup, or something in between. Whatever you decide, I hope you get to know your manager well enough to know that they have your back, your team well enough to know that you like working with them, and your project well enough to know that you find it interesting. Personally, I'm a bit tired of the sort of nonsense you see at big companies after two stints at big companies, and I might want to trade that for <a href="http://totalgarb.tumblr.com/tagged/startupbullshit">the sort of nonsense you see at startups</a> next time I look for work, but that's just me. You should figure out what the relevant tradeoffs are for you.</p>
<p><a href="https://danluu.com/startup-options/">Myself on options vs. cash</a>.</p>
<p><a href="https://web.archive.org/web/20141122200442/https://medium.com/@jocelyngoldfein/the-innovation-dead-end-ee25f8aa090f">Jocelyn Goldfein on big companies vs. small companies</a>.</p>
<p><a href="http://www.kalzumeus.com/2011/10/28/dont-call-yourself-a-programmer/">Patrick McKenzie on providing business value vs. technical value</a>, with <a href="http://yosefk.com/blog/do-call-yourself-a-programmer-and-other-career-advice.html">a response from Yossi Kreinin</a>.</p>
<p><a href="http://yosefk.com/blog/do-you-really-want-to-be-making-this-much-money-when-youre-50.html">Yossi Kreinin on passion vs. money</a>, and with a <a href="http://yosefk.com/blog/stock-options-a-balanced-approach.html">rebuttal to this post on regret minimization</a>.</p>
<p><em>Update: The responses on this post have been quite divided. Folks at big companies usually agree, except that the numbers seem low to them, especially for new grads. This is true even for people who living in places like Madison and Austin, which have a cost of living similar to U.S. median. On the other hand, a lot of people vehemently maintain that the numbers in this post are basically impossible. A lot of people are really invested in the idea that they're making about as much as possible. If you've decided that making less money is the right trade-off for you, that's fine and I don't have any problem with that. But if you really think that you can't make that much money and you don't believe me, I recommend talking to one of the hundreds of thousands of engineers at one of the many large companies that pays well.</em></p>
<p><em>Update 2: This post was originally written in 2015, when the $250k number would be conservative but not unreasonable for someone who's "senior" at Google or FB. If we look at the situation today in 2017, people <a href="https://twitter.com/danluu/status/942452212445405185">one entire band below that are regularly bringing in $250k</a> and a better estimate might be $300k or $350k. I'm probably more bear-ish on future dev compensation than most people, but things are looking pretty good for now and an event that wipes out big company dev compensation seems likely to do even worse things to the options packages for almost all existing startups.</em></p>
<p><small>Thanks to Kelly Eskridge, Leah Hanson, Julia Evans, Alex Clemmer, Ben Kuhn, Malcolm Matalka, Nick Bergson-Shilcock, Joe Wilder, Nat Welch, Darius Bacon, Lindsey Kuper, Prabhakar Ragde, Pierre-Yves Baccou, David Turner, Oskar Thoren, Katerina Barone-Adesi, Scott Feeney, Ralph Corderoy, Ezekiel Benjamin Smithburg, and Kyle Littler for comments/corrections/discussion.</small></p>
</div>
<br>
<br>
<br>
<br>
<br>
<br>
via danluu.com https://danluu.com<br>
<br>
August 2, 2019 at 12:08PM
